### PR TITLE
server: refresh expired store address on resolve

### PR DESCRIPTION
**Pull request by huachaohuang**
Reference Issue: #1153

### CODE REVIEW

File: src/server/resolve.rs [hunk 1]
```diff
@@ -44,45 +47,53 @@ impl Display for Task {
...
-        Ok(s)
+        let s = box_try!(pd_client.get_store(store_id));
+        if s.get_state() == metapb::StoreState::Tombstone {
+            RESOLVE_STORE_COUNTER.with_label_values(&["tombstone"]).inc();
+            return Err(box_err!("store {} has been removed", store_id));
+        }
+        let addr = s.get_address().to_owned();
+        // In some tests, we use empty address for store first,
+        // so we should ignore here.
+        // TODO: we may remove this check after we refactor the test.
```
Comment (hhkbp2) [inline src/server/resolve.rs]: TODO should be removed?
A health check would be necessary.

File: src/server/resolve.rs [hunk 2]
```diff
@@ -168,7 +187,12 @@ mod tests {
             unimplemented!();
         }
         fn get_store(&self, _: u64) -> Result<metapb::Store> {
-            Ok(self.store.clone())
+            // The store address will changed every second.
```
Comment (hhkbp2) [inline src/server/resolve.rs]: s/changed/change on/

If the address change every 10ms, it will make the test case below run faster.

**Comment by hhkbp2** [comment]
The CI fails on `test raftstore::test_multi::test_multi_server_random_latency`. It's caused by the high package delay/drop rate used by `RandomLatencyFilter`, so that there is no leader elected for the test cluster.
Please change the delay/drop rate from 50 to 20 so that the leader could be elected in a reasonable time.
https://github.com/pingcap/tikv/blob/master/tests/raftstore/test_multi.rs#L209

File: src/server/resolve.rs (hunk 1 above)
Comment (huachaohuang) [inline src/server/resolve.rs]: The TODO hasn't been resolved yet, we should keep it.

**Comment by huachaohuang** [comment]
It's not related to this PR, we should fix it in another PR.

**Comment by hhkbp2** [comment]
LGTM

**Comment by siddontang** [comment]
LGTM

**Pull request merged**